ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useEditorSearch.ts

```diff
@@ -0,0 +1,151 @@
+import { useCallback, useEffect, useState } from 'react';
+// Helper functions that use the cursor
+export default function useCursorUtils(CodeMirror: any) {
+
+	const [markers, setMarkers] = useState([]);
+	const [overlay, setOverlay] = useState(null);
+	const [scrollbarMarks, setScrollbarMarks] = useState(null);
+	const [overlayTimeout, setOverlayTimeout] = useState(null);
+	const [previousKeywordValue, setPreviousKeywordValue] = useState(null);
+
+	const clearMarkers = useCallback(() => {
+		for (let i = 0; i < markers.length; i++) {
+			markers[i].clear();
+		}
+
+		setMarkers([]);
+	}, [markers]);
+
+	const clearOverlay = useCallback((cm: any) => {
+		if (overlay) cm.removeOverlay(overlay);
+		if (scrollbarMarks) scrollbarMarks.clear();
+		if (overlayTimeout) clearTimeout(overlayTimeout);
+
+		setOverlay(null);
+		setScrollbarMarks(null);
+		setOverlayTimeout(null);
+	}, [overlay, scrollbarMarks, overlayTimeout]);
```

Comment:
As above this will be captured in defineExtension so might deal with stale state. I think you can use the same technique of functional updates again.

Another way to prevent stale state is to use `useReducer`. Although I don't think you need it here, and I haven't used it at all either, but [this SO answer](https://stackoverflow.com/a/60460551/561309) made me realise it's a good way to avoid stale state in closures.